pkg/scheduler/core/equivalence_cache.go

```diff
@@ -45,7 +42,7 @@ type EquivalenceCache struct {
 // The AlgorithmCache stores PredicateMap with predicate name as key
 type AlgorithmCache struct {
 	// Only consider predicates for now
-	predicatesCache *lru.Cache
+	predicatesCache *cacheutil.Cache
```

Comment:
cacheutil.Cache is a wrapper around a map, and we have two other maps, algorithmCache and PredicateMap. I would suggest using plain maps for all three rather than implementing a simple cache.

(Alternatively, all three could use cacheutil.Cache, but that involves a lot of type assertions.)